### PR TITLE
[CSharpBinding] Added throttling logic to updating ProjectCatalog on …

### PR DESCRIPTION
**Pull request by DavidKarlas**
…document change to limit CPU consumption

/cc: @mkrueger @Therzok

### CODE REVIEW

**Comment by Therzok** [comment]
Lgtm!

File: main/src/addins/CSharpBinding/MonoDevelop.CSharp/ProjectSearchCategory.cs [hunk 1]
```diff
@@ -250,9 +250,30 @@ public void Dispose ()
...
+								documentChangedCts [e.DocumentId] = tcs;
+							}
+							try {
+								//Delaying parsing of new content for 1 second shouldn't be noticable by user
+								//since he would have to edit file and instantlly go to search for newly written member...
+								await Task.Delay (1000, tcs.Token).ConfigureAwait (false);
+								await Task.Run (delegate {
+									return UpdateDocument (documentInfos, doc, tcs.Token);
+								}, tcs.Token).ConfigureAwait (false);
+							} finally {
```
Comment (alanmcgovern) [inline main/src/addins/CSharpBinding/MonoDevelop.CSharp/ProjectSearchCategory.cs]: You'll need a `catch (OperationCancelledException) { }` in here somewhere!

Comment (DavidKarlas) [inline main/src/addins/CSharpBinding/MonoDevelop.CSharp/ProjectSearchCategory.cs]: Expand down, there is 
`} catch (OperationCanceledException) {`
and even
```} catch (AggregateException ae) {
 					ae.Flatten ().Handle (ex => ex is OperationCanceledException);```

Comment (DavidKarlas) [inline main/src/addins/CSharpBinding/MonoDevelop.CSharp/ProjectSearchCategory.cs]: Hence I added `finnaly`...

Comment (alanmcgovern) [inline main/src/addins/CSharpBinding/MonoDevelop.CSharp/ProjectSearchCategory.cs]: 👍

**Comment by Therzok** [comment]
Ready to merge?

**Pull request merged**